### DISCUSSION TITLE
pytorch loader -> which branch?

### DISCUSSION THREAD

**Discussion by luisoala** [mlcommons/croissant#406] [Q&A] [answered]
i remembered someone worked on a pt adapter to load croissant into pt dataset object. searching the issues and branches and couldnt find it. any hints?

**Comment by mkuchnik** [comment, marked as answer]
@luisoala There is currently a minimal [notebook](https://github.com/mlcommons/croissant/pull/444#issue-2065961858) for an adapter using FLORES-200. I am planning to expand on the core functionality of the adapter in future patches. Some of the obvious pain points are 1) multiprocessing and 2) handling of complex object types (e.g., images, video), which have their own semantics and performance issues. Let me know if you have any thoughts :)

**Comment by luisoala** [comment]
thx michael! ill take a look at it next week. that looks promising. let me know if i can support, im interested, also in the performance direction

**Comment by marcenacp** [comment]
FYI, the [TFDS recipe](https://github.com/mlcommons/croissant/blob/main/python/mlcroissant/recipes/tfds_croissant_builder.ipynb) also uses a DataLoader with pre-prepared data and showcases an end-to-end training with torch and torch.data.
